service/src/componentTest/kotlin/com/integreety/activityengine/step/FindActivityApiSteps.kt

```diff
@@ -0,0 +1,68 @@
+package com.integreety.activityengine.step
+
+import com.googlecode.yatspec.state.givenwhenthen.TestState
+import com.integreety.activityengine.ActivityEngineApplication
+import com.integreety.activityengine.api.request.ActivityRequest
+import com.integreety.activityengine.api.response.ActivityResponse
+import com.integreety.activityengine.client.ActivityEngineClient
+import com.integreety.activityengine.config.SequenceDiagramConfig
+import io.cucumber.java8.En
+import io.cucumber.spring.CucumberContextConfiguration
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.*
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment.DEFINED_PORT
+import org.springframework.boot.test.web.client.TestRestTemplate
+import org.springframework.cloud.openfeign.EnableFeignClients
+import org.springframework.context.annotation.Import
+import org.springframework.http.ResponseEntity
+import org.springframework.test.context.TestPropertySource
+
+@CucumberContextConfiguration
+@SpringBootTest(webEnvironment = DEFINED_PORT, classes = [ActivityEngineApplication::class])
+@EnableFeignClients(clients = [ActivityEngineClient::class])
+@TestPropertySource("classpath:application-test.properties")
+@Import(SequenceDiagramConfig::class)
+class FindActivityApiSteps(
+    val activityEngineClient: ActivityEngineClient,
+    val testRestTemplate: TestRestTemplate,
+    val testState: TestState
+) : En {
+
+    private var activityResponse: ResponseEntity<ActivityResponse>? = null
+    private lateinit var activityResponses: ResponseEntity<List<ActivityResponse>>
+
+    init {
+
+        // TODO Is there a better way to do it? (This is to exclude the test calls from LSD)
+        testRestTemplate.restTemplate.interceptors.clear()
+
+        Given("an existing activity with lessonId {word}") { lessonId: String ->
+            activityResponse = testRestTemplate.postForEntity(
+                "/activities",
+                ActivityRequest(lessonId = lessonId),
+                ActivityResponse::class.java
+            )
+            testState.interestingGivens().add("lessonId", lessonId)
+        }
+
+        Given("a non existent lessonId") {
+        }
+
+        When("the activity is requested by the returned id") {
+            activityResponse = activityResponse?.body?.let { activityEngineClient.find(it.id) }
+        }
+
+        When("the activity is requested by lessonId {word}") { lessonId: String ->
+            activityResponses = activityEngineClient.findByLessonId(lessonId)
+        }
+
+        Then("the activity is returned with lessonId {word}") { lessonId: String ->
+            assertThat(activityResponse?.body?.lessonId, equalTo(lessonId))
+        }
+
+        Then("empty list is returned") {
+            assertThat(activityResponses.body, `is`(empty()))
```

Comment:
shame about the back-ticks (guessing they are needed here?)

Reply:
`is` is a Kotlin keyword so this is the way to escape it.